.github/workflows/release.yml

```diff
@@ -9,11 +9,19 @@ on:
       - 'V*.*.*'
 jobs:
   release:
-    runs-on: ubuntu-latest
+    runs-on: 
+      - self-hosted
+      - X64
+      - Linux
+      - PSoC
+
     if: startsWith(github.ref, 'refs/tags/V')
     steps:
-    - uses: actions/checkout@v3
-    - uses: actions/setup-python@v4
+    - name: Checkout actions
```

Comment:
This is secondary, but we should restrict more the release triggers. Only tags and or publish? 
Generally the release will come from the "main", and all the ci/cd should be green. 
Eventually some interim dev release would come from other branches, like right now. Maybe also handle those exceptions